src/orion/algo/base.py

```diff
@@ -310,9 +310,11 @@ def has_observed(self, trial):
             True if the trial's objective was observed by the algo, False otherwise.
 
         """
-        return (
-            self.get_id(trial) in self._trials_info
-            and self._trials_info[self.get_id(trial)][1] is not None
+        if not self.has_suggested(trial):
+            return False
+        return self._trials_info[self.get_id(trial)][0].status in (
```

Comment:
Why does this check only the first index? Is it always a len(1) sequence? If there were to be duplicates, should this do anything different?

Reply:
It's a bad design. I'd like to address this with a more formal in-memory storage for algorithms are shortly described here: https://github.com/Epistimio/orion/issues/802. 

The format is
 ```python
self._trials_info[some_trial_id] = (trial_object, trial_objective)
```

Previously we only looked at the objective at index 1, now we only look at the trial at index 0.